legacy/jacorb/src/main/java/org/jboss/as/jacorb/MigrateOperation.java

```diff
@@ -70,62 +103,123 @@ public void execute(final OperationContext context, final ModelNode operation) t
             throw new OperationFailedException("the iiop migration can be performed when the server is in admin-only mode");
         }
 
-        final PathAddress subsystemsAddress=context.getCurrentAddress().getParent();
+        final PathAddress subsystemsAddress = context.getCurrentAddress().getParent();
 
         if (context.readResourceFromRoot(subsystemsAddress).hasChild(OPENJDK_SUBSYSTEM_ELEMENT)) {
             throw new OperationFailedException("can not migrate: the new iiop-openjdk subsystem is already defined");
         }
 
-        if(!context.readResourceFromRoot(PathAddress.EMPTY_ADDRESS).hasChild(OPENJDK_EXTENSION_ELEMENT)) {
-            addOpenjdkExtension(context);
-        }
+        final Map<PathAddress, ModelNode> migrateOperations = new LinkedHashMap<>();
 
-        final Resource jacorbResource = context.readResourceForUpdate(PathAddress.EMPTY_ADDRESS);
-        final ModelNode jacorbModel = Resource.Tools.readModel(jacorbResource).clone();
+        if (!context.readResourceFromRoot(PathAddress.EMPTY_ADDRESS).hasChild(OPENJDK_EXTENSION_ELEMENT)) {
+            addOpenjdkExtension(context, migrateOperations);
+        }
 
-        TransformUtils.checkLegacyModel(jacorbModel, true);
+        context.addStep(new OperationStepHandler() {
+            @Override
+            public void execute(OperationContext operationContext, ModelNode modelNode) throws OperationFailedException {
 
-        final ModelNode openjdkModel = TransformUtils.transformModel(jacorbModel);
+                final Resource jacorbResource = context.readResourceForUpdate(PathAddress.EMPTY_ADDRESS);
+                final ModelNode jacorbModel = Resource.Tools.readModel(jacorbResource).clone();
+
+                List<String> unsupportedProperties = TransformUtils.checkLegacyModel(jacorbModel);
+                if (!unsupportedProperties.isEmpty()) {
+                    context.setRollbackOnly();
+
+                    ModelNode result = new ModelNode();
+                    final String error =JacORBLogger.ROOT_LOGGER.cannotEmulatePropertiesWarning(unsupportedProperties);
```

Comment:
Should be be immediatly failing here? I think it would potentially be a lot more user friendly to just report this an warning in a migration-warnings section (see web and JMS for an example).